server/src/main/java/org/apache/iotdb/db/mpp/execution/operator/source/SeriesAggregationScanOperator.java

```diff
@@ -96,337 +57,6 @@ public SeriesAggregationScanOperator(
-      calcFromBatch(tsBlock, curTimeRange);
-
-      // judge whether the calculation finished
-      boolean isTsBlockOutOfBound =
-          ascending
-              ? tsBlock.getEndTime() > curTimeRange.getMax()
-              : tsBlock.getEndTime() < curTimeRange.getMin();
-      if (isEndCalc(aggregators) || isTsBlockOutOfBound) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  private boolean readAndCalcFromChunk(TimeRange curTimeRange) throws IOException {
-    while (seriesScanUtil.hasNextChunk()) {
-      if (canUseCurrentChunkStatistics()) {
-        Statistics chunkStatistics = seriesScanUtil.currentChunkStatistics();
-        if (chunkStatistics.getStartTime() > curTimeRange.getMax()) {
-          if (ascending) {
-            return true;
-          } else {
-            seriesScanUtil.skipCurrentChunk();
-            continue;
-          }
-        }
-        // calc from chunkMetaData
-        if (curTimeRange.contains(chunkStatistics.getStartTime(), chunkStatistics.getEndTime())) {
-          calcFromStatistics(chunkStatistics);
-          seriesScanUtil.skipCurrentChunk();
-          if (isEndCalc(aggregators) && !isGroupByQuery) {
-            return true;
-          } else {
-            continue;
-          }
-        }
-      }
-
-      // read page
-      if (readAndCalcFromPage(curTimeRange)) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  private void calcFromStatistics(Statistics statistics) {
-    for (int i = 0; i < aggregators.size(); i++) {
-      Aggregator aggregator = aggregators.get(i);
-      if (aggregator.hasFinalResult()) {
-        continue;
-      }
-      aggregator.processStatistics(statistics);
-    }
-  }
-
-  public boolean canUseCurrentFileStatistics() throws IOException {
-    Statistics fileStatistics = seriesScanUtil.currentFileStatistics();
-    return !seriesScanUtil.isFileOverlapped()
-        && containedByTimeFilter(fileStatistics)
-        && !seriesScanUtil.currentFileModified();
-  }
-
-  public boolean canUseCurrentChunkStatistics() throws IOException {
-    Statistics chunkStatistics = seriesScanUtil.currentChunkStatistics();
-    return !seriesScanUtil.isChunkOverlapped()
-        && containedByTimeFilter(chunkStatistics)
-        && !seriesScanUtil.currentChunkModified();
-  }
-
-  public boolean canUseCurrentPageStatistics() throws IOException {
-    Statistics currentPageStatistics = seriesScanUtil.currentPageStatistics();
-    if (currentPageStatistics == null) {
-      return false;
-    }
-    return !seriesScanUtil.isPageOverlapped()
-        && containedByTimeFilter(currentPageStatistics)
-        && !seriesScanUtil.currentPageModified();
-  }
-
-  private boolean containedByTimeFilter(Statistics statistics) {
-    Filter timeFilter = seriesScanUtil.getTimeFilter();
-    return timeFilter == null
-        || timeFilter.containStartEndTime(statistics.getStartTime(), statistics.getEndTime());
+    this.subSensorSize = 1;
```

Comment:
put this field into super's constructor.

Reply:
fixed